README.md

```diff
@@ -797,6 +797,82 @@ In either case:
     end
     ```
 
+### Nested conditionals
+
+* <a name="no-nested-conditionals"></a>
+  Avoid the use of nested conditionals for flow of control. 
+  ([More on this][avoid-else-return-early].) <sup>[[link](#no-nested-conditionals)]</sup>
+
+  Prefer a guard clause when you can assert invalid data. A guard clause
+  is a conditional statement at the top of a function that bails out as
+  soon as it can. 
+  
+  The general principles boil down to:
+  * Return as soon as you know your method cannot do any more meaningful work
+  * Reduce indentation by using `if`/`unless`/`return` instead of a top-level 
+  `if`/`else`
+  * Try keep the “meat” of your method at the lowest indentation level.
+
+  ```ruby
+  # bad
+  def compute_thing(thing)
+    if thing[:foo]
+      update_with_bar(thing[:foo])
+      if thing[:foo][:bar]
+        partial_compute(thing)
+      else
+        re_compute(thing)
+      end
+    else
+      []
+    end
+  end
+
+  # good
+  def compute_thing(thing)
+    return [] unless thing[:foo]
+    update_with_bar(thing[:foo])
+    return re_compute(thing) unless thing[:foo][:bar]
+    partial_compute(thing)
+  end
+  ```
+
+  Prefer `next` in loops instead of conditional blocks.
+
+  ```ruby
+  # bad
+  [0, 1, 2, 3].each do |item|
+    if item > 1
+      puts item
+    end
+  end
+
+  # good
+  [0, 1, 2, 3].each do |item|
+    next unless item > 1
+    puts item
+  end
+  ```
+
+  The same technique can be used to untangle complex booleans. 
+  
+  ```ruby
+  # bad
+  def should_do_something?
+    condition_a && (((condition_b || condition_c) && condition_d) || condition_e)
```

Comment:
Hmm.. good point. I like the local variables, but I also want us to return early to avoid doing a bunch of unnecessary computation. A good example is a ```is_valid?``` method which can return ```false``` if a bunch of fields aren't valid. It doesn't need to do ```a_valid && b_valid && c_valid && d_valid && e_valid ```.

Do you feel such an example might be more clear? I'm trying to avoid using actual production code which is making it harder to come with an example.